### ISSUE TITLE
STK: could stk provide an option to call Teuchos::TimeMonitor::summarize()?

### ISSUE THREAD

**Issue by jhux2** [trilinos/Trilinos#10927]
## Question
@trilinos/stk
@tasmith4 

Would it be possible for STK to provide an option to call Teuchos::TimeMonitor::summarize()?   This would be useful for Trilinos developers working with Sierra apps. In particular, they could request detailed Trilinos timer output without having to modify the application code.

**Comment by alanw0** [comment]
@jhux2 Hi Jonathan, can you provide a little more detail? Would you like this to be called around the time of MPI_finalize? STK doesn't currently wrap MPI_finalize or Kokkos_finalize, but the subject has come up recently...

**Comment by jhux2** [comment]
Hi Alan, this came about from a discussion with @rstumin.  Please correct me if I'm wrong, but my understanding is that Sierra apps typically call STK to get memory and timing information at the end of a simulation.  I was thinking that whatever call that is might also be able to request a report of any Teuchos timers.  This would presumably happen in `main.cpp`, but before finalizing MPI.

**Comment by jhux2** [comment]
Of course, the Trilinos developer could always add the `summarize` call to the particular app they're working with.  I was thinking that perhaps this could be done once and then be available across all apps that use STK timers.

**Comment by alanw0** [comment]
ok, yes let me look into this to see what the possibilities are. the stk module where that time/memory table is produced (stk_util) does already depend on TeuchosCore so this probably wouldn't be very difficult.
What if you could do something like this:
"mpirun ... aria ... --output-teuchos-timers"
Or something like that?

**Comment by jhux2** [comment]
@alanw0 Yes, that would do the trick.  I'll let @rstumin weigh in, as well.

**Comment by rstumin** [comment]
Alanw0 @jhu2 Sorry to be so late ... somehow this slipped through my filter. Yes, this would be perfect. I recently ran into this issue with Aria where I basically wanted the timer info at the end of the entire simulation. As Jonathan indicated, I just modified the application so that it did the summarize() call at the right spot. It wasn't a totally big deal but sometimes I'm not even compiling a sierra app and just using a pre-existing binary ... where it would be nice to get Trilinos timing information without having to compile.

**Comment by jhux2** [comment]
Would still be nice to have.

**Comment by alanw0** [comment]
@jhux2 sorry this slipped through the cracks!
I will try to get this done right away. It's a sierra-framework change, technically. I'll lodge a sierra-help ticket for the stk team, which will reduce the likelihood of it slipping away again.

**Comment by jhux2** [comment]
@alanw0 Thanks, and no worries!
